load-testing/tests/relays_stress_helpers_test.go

```diff
@@ -1034,150 +1031,189 @@ func (s *relaysSuite) sendRelay(iteration uint64, relayPayload string) (appAddre
 	gateway := s.activeGateways[iteration%uint64(len(s.activeGateways))]
 	application := s.activeApplications[iteration%uint64(len(s.activeApplications))]
 
-	gatewayUrl, err := url.Parse(s.gatewayUrls[gateway.address])
-	require.NoError(s, err)
-
-	// Include the application address in the query to the gateway.
-	query := gatewayUrl.Query()
-	query.Add("applicationAddr", application.address)
-	query.Add("relayCount", fmt.Sprintf("%d", iteration))
-	gatewayUrl.RawQuery = query.Encode()
-
-	// Use the pre-defined service ID that all application and suppliers are staking for.
-	gatewayUrl.Path = testedServiceId
-
 	// TODO_MAINNET: Capture the relay response to check for failing relays.
 	// Send the relay request within a goroutine to avoid blocking the test batches
 	// when suppliers or gateways are unresponsive.
-	go func(gwURL, payload string) {
-		_, err = http.DefaultClient.Post(
-			gwURL,
-			"application/json",
-			strings.NewReader(payload),
-		)
+	go func(gwURL, appAddr, payload string) {
+		gwURL = gwURL + "/"
+		req, err := http.NewRequest("POST", gwURL, strings.NewReader(payload))
+
+		req.Header.Add("X-App-Address", appAddr)
+		res, err := http.DefaultClient.Do(req)
 		require.NoError(s, err)
-	}(gatewayUrl.String(), relayPayload)
+
+		if res.StatusCode != http.StatusOK {
+			s.failedRelays.Add(1)
+		}
+	}(s.gatewayUrls[gateway.address], application.address, relayPayload)
```

Comment:
What if you do something like:

```go
inlineFunc := func(...) {..}
gatewayURL := s.gatewayUrls[...]
go inlineFunc(gatewayURL, ...)
```

You'll need better names, but it'll be easier to maintain.,